### PR TITLE
inference: Surface lua `print(message)` to UI

### PR DESCRIPTION
**Pull request by efritz**
Feed the lua sandbox `print` output back to the UI in the inference page.

image

## Test plan

Tested locally.

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.22 kb) | 0.00% (+0.22 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ef/sandbox-print/compare-report.html;tab=live_object) for a full comparison between the commits d2dfd31 and 39999bb or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: enterprise/internal/codeintel/autoindexing/service.go [hunk 1]
```diff
@@ -80,52 +80,43 @@ func (s *Service) GetIndexConfigurationByRepositoryID(ctx context.Context, repos
 
 // InferIndexConfiguration looks at the repository contents at the latest commit on the default branch of the given
 // repository and determines an index configuration that is likely to succeed.
-func (s *Service) InferIndexConfiguration(ctx context.Context, repositoryID int, commit string, localOverrideScript string, bypassLimit bool) (_ *config.IndexConfiguration, _ []config.IndexJobHint, err error) {
+func (s *Service) InferIndexConfiguration(ctx context.Context, repositoryID int, commit string, localOverrideScript string, bypassLimit bool) (_ *config.IndexConfiguration, _ string, err error) {
```
Comment (keynmol) [inline enterprise/internal/codeintel/autoindexing/service.go:83]: May be it's worth merging the log output along with config.IndexConfiguration into some new struct, like `InferenceResult` - and do similarly for other cases where you needed to add a third return parameter?

Just to keep things a bit tidier

Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/service.go]: Updated.

File: enterprise/internal/codeintel/autoindexing/internal/inference/service.go [hunk 2]
```diff
@@ -183,26 +188,32 @@ func (s *Service) inferIndexJobOrHints(
 	commit string,
 	overrideScript string,
 	invocationContextMethods invocationFunctionTable,
-) ([]indexJobOrHint, error) {
+) (_ []indexJobOrHint, logs string, _ error) {
```
Comment (keynmol) [inline enterprise/internal/codeintel/autoindexing/internal/inference/service.go:191]: nit: I'd wrap this string and results in a separate struct as well

Up to you, can be done at a later point if gets out of hand.

**Review by keynmol — APPROVED** [review]

**Pull request merged**